### DISCUSSION TITLE
明确回复消息的行为

### DISCUSSION THREAD

**Discussion by Anillc** [howmanybots/onebot#125] [想法 / 建议]
从某个版本开始，qq 中回复消息会自带一个 at ，这个 at 肯定不会是大多数 onebot 实现和使用者想要的。也许应该**明确**一下，onebot 实现是否应该处理平台发过来的多余的 at

当然，我认为这个 at 不应该由 onebot 使用方处理，而应该由 onebot 实现来处理

## 举例

在 go-cqhttp 中

这样的回复和收到的消息分别为

![image](https://user-images.githubusercontent.com/23411248/134040224-d8d2167e-f7c5-4bab-8f33-203b89a3b3a7.png)
`[CQ:reply,id=-123456][CQ:at,qq=123456] 2`

![image](https://user-images.githubusercontent.com/23411248/134040523-fde88974-5e95-446d-b7a3-594ad081a6b5.png)
`[CQ:reply,id=123456][CQ:at,qq=123456] [CQ:at,qq=123456] 2`

**Comment by stdrc** [comment]
听起来非常有道理，我个人感觉 OneBot 实现应该把多余的 `at` 去掉，可以考虑在 `reply` 里加上一个参数叫 `user_id` 来表示回复的是谁的消息

cc @crazywhalecc

**Comment by ExerciseBook** [comment]
我觉得这个是 txsb。
这个特性据说是由于部分平台的 qq 实现不会监听回复，导致后来某个版本的 QQ 变成了回复加 At。很他妈怪。

https://github.com/ParaParty/OneBot-Framework/blob/main/OneBot-CommandRoute/CommandRoute/Lexer/CommandLexer.cs#L85
这个傻逼特性害我多写了一堆屎山。



问题就在，如果去掉这个 at，那么会不会产生其他奇怪的问题？
